llvm/include/llvm/CodeGen/TailDuplication.h

```diff
@@ -0,0 +1,47 @@
+//===- llvm/CodeGen/TailDuplication.h ---------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CODEGEN_TAILDUPLICATIONPASS_H
+#define LLVM_CODEGEN_TAILDUPLICATIONPASS_H
+
+#include "llvm/CodeGen/MBFIWrapper.h"
+#include "llvm/CodeGen/MachineFunction.h"
+#include "llvm/CodeGen/MachinePassManager.h"
+
+namespace llvm {
+
+template <typename DerivedT, bool PreRegAlloc>
+class TailDuplicatePassBase : public PassInfoMixin<DerivedT> {
+private:
+  std::unique_ptr<MBFIWrapper> MBFIW;
```

Comment:
IIUC `MBFIWrapper` exists here because legacy pass manager can't compute analysis results lazily. In new pass manager we can just use `MFAM.getResult<MachineBlockFrequencyAnalysis>()`, all results are computed lazily.

Reply:
`TailDuplicator` is  using `MBFIWrapper`, so I'll have to change there

Reply:
MBFIWrapper changes have a different scope, should I try to remove it from this pass in a separate pr?

Reply:
Oh, I see the `TailDuplicator`, we can remove `MBFIWrapper ` in future.